src/modules/complianceengine/src/lib/FilesystemScanner.cpp

```diff
@@ -0,0 +1,423 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "FilesystemScanner.h"
+
+#include "Optional.h"
+
+#include <algorithm>
+#include <cstdio>
+#include <cstdlib>
+#include <dirent.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <fstream>
+#include <limits.h>
+#include <memory>
+#include <signal.h>
+#include <sstream>
+#include <stack>
+#include <string.h>
+#include <sys/file.h>
+#include <sys/stat.h>
+#include <sys/statfs.h>
+#include <unistd.h>
+#include <vector>
+
+namespace ComplianceEngine
+{
+namespace
+{
+
+class FileLock
+{
+public:
+    explicit FileLock(const std::string& path)
+        : m_path(path),
+          m_fd(::open(m_path.c_str(), O_RDWR | O_CREAT, 0644))
+    {
+
+        if (m_fd < 0)
+        {
+            return;
+        }
+        if (::flock(m_fd, LOCK_EX | LOCK_NB) != 0)
+        {
+            ::close(m_fd);
+            m_fd = -1;
+            return;
+        }
+        if (ftruncate(m_fd, 0) == 0)
+        {
+            char pidBuf[64];
+            int len = std::snprintf(pidBuf, sizeof(pidBuf), "%ld\n", (long)::getpid());
+            if (len > 0)
+            {
+                (void)::write(m_fd, pidBuf, static_cast<size_t>(len));
+                ::lseek(m_fd, 0, SEEK_SET);
+            }
+        }
+    }
+    ~FileLock()
+    {
+        if (m_fd >= 0)
+        {
+            ::flock(m_fd, LOCK_UN);
+            ::close(m_fd);
+            m_fd = -1;
+        }
+    }
+    bool IsLocked() const
```

Comment:
nit: our coding standard asks that {} blocks are separated by one blank line. Applies to multiple places.